modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/model/ExtensionTypeFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.introspection.describer.model;
+
+import org.mule.runtime.extension.api.annotation.Extension;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.runtime.ExtensionTypeWrapper;
+
+import javax.lang.model.element.TypeElement;
+
+/**
+ * Method Factory Pattern implementation that creates {@link ExtensionElement} from {@link Class} or {@link TypeElement}
+ *
+ * @since 4.0
+ */
+public class ExtensionTypeFactory
```

Comment:
I'll accept this for now, but this class will become useless when you tackle on the AST part. Let's resume this discussion when you start that issue